File: neal-sharma/day-04/century.js
/* precept
parameters- century 1 spans from 1-100, century 2 spans from 101-200 etc.
return value- return the century.
example- 1705 = 18th century 1900 = 19th century
concerns- if loop? method?
explain- Math.ceil rounds the number up to the nearest integer if necessary and returns the result
positive self talk- I can figure it just like i have for the rest of them so far.
translate- Math.ceil-round up number to nearest integer- (year/100) (1705/100) = 17.05 rounded up
to nearest integer = 18
*/

function century(year) {
  return Math.ceil(year / 100);
}

/*
  Ember's Feedback:
  Yep excellent job!
  A concern here would be what is your expected failure behavior.
  At the moment, is you are given a negative number it will return using your logic,
  but maybe if you were writing this function for yourself you would want to return something
  like "B.C.E. unsupported" or something.
*/

File: neal-sharma/day-31/century.py
'''
Parameters- centuries. year 1 - 100 = century 1, second = 101 - 200
return- given a year, return the century that it belongs to.
example- 1705 = 18
explain- This one was pretty simple. take the year, divide by 100, return the math.ceil of
that number. just dividing by 100 would give you ex. 17.05 using math.ceil we round up to 18.
Positive self talk- Keep your face to the sunshine and you cannot see a shadow. Helen Keller
'''

import math


def century(year):
    return math.ceil(year/100)


"""
    Ember's Feedback:
    - Great job :D
    - 🌻
"""


# Alternative Solutions
def century(year):
    return year // 100 + (1 if year % 100 != 0 else 0)


def century(year):
    return (year + 99) // 100


def century(year):
    return (year - 1) // 100 + 1
